mill-scalafix/src/com/goyeau/mill/scalafix/ScalafixModule.scala

```diff
@@ -77,8 +76,8 @@ object ScalafixModule {
       wd: os.Path
   ): Result[Unit] =
     if (sources.nonEmpty) {
-      val scalafix = Scalafix
-        .fetchAndClassloadInstance(scalaVersion, repositories.map(CoursierUtils.toApiRepository).asJava)
+      val scalafix = ScalafixCache
+        .getOrElseCreate(scalaVersion, repositories)
```

Comment:
Potential tweak even further: as I was mentioning [no later than yesterday in the docs](https://github.com/scalacenter/scalafix/pull/2087/files), only a major.minor version is required, so FWIW you could strip the patch version to maximize cache hit ratio (the instance will be the same for 3.3.3 and 3.3.4).

Reply:
That assumption is true now but it might become false in future versions of scalafix. Since it's very rare to use different patch versions in the same build, I wouldn't bother trying to be clever here.

Reply:
Alternatively, we could use the full scalafix classpath (as `PathRef`s, either sorted or as `Set`) as cache key.